BurnOutSharp/ProtectionType/DiscGuard.cs

```diff
@@ -1,23 +1,168 @@
+    /// Due to the fact that these games can seemingly be burned to CD-R under the right conditions using CloneCD, it likely isn't using twin sectors or DPM (https://www.gameburnworld.com/protections_discguard.shtml).
+    /// It seems likely to be using subchannels as the basis for this signature, as no errored sectors are used.
+    /// DiscGuard is seemingly able to be detect so-called "active debuggers", based on text found in "Alternate.exe" (Redump entry 31914) and "Alt.exe" (Redump entries 46743, 46961, 79284, and 79374).
+    /// There's a reference to a file "Dg.vbn" being present in "Alternate.exe" (Redump entry 31914) and "Alt.exe" (Redump entries 46743, 46961, 79284, and 79374), but no copy has been found in any sample so far.
+    /// There seem to be two distinct versions of DiscGuard, with one only being present on one known game (Redump entry 31914).
+    /// "Alternate.exe" in the outlying version is "Alt.exe" in the rest.
+    /// "TTR1.DLL" and "TTR2.DLL" in the outlying version appear to be "T111.DLL" and "T222.DLL" in the rest, respectively.
+    /// "IOSLink.sys" and "IOSLink.vxd" seem to be the same throughout all versions, even down to the reported File Version in "IOSLink.sys".
+    /// There are other various DLLs, but their names seem to be different on every disc. Their filenames all seem to start with the letter "T", and it seems to be one of these specifically that the protected executable imports.
+    /// These outlying DLLs are:
+    /// "T29.dll" (Redump entry 31914).
+    /// "T5375.dll" (Redump entry 79284).
+    /// "TD352.dll" and "TE091.dll" (Redump entry 46743).
+    /// "T71E1.dll" and "T7181.dll" (Redump entry 46961).
+    /// "TA0E4.DLL" (Redump entry 79374).
+    /// </summary>
+    public class DiscGuard : IPathCheck, IPortableExecutableCheck
     {
+        // TODO: Add checks for the game executables, which seem likely to contain some kind of indicators that can be checked for. The game executables all seem to import "Ord(1)" from one of the varying DLLs present.
+        /// <inheritdoc/>
+        public string CheckPortableExecutable(string file, PortableExecutable pex, bool includeDebug)
+        {
+            // Get the sections from the executable, if possible
+            var sections = pex?.SectionTable;
+            if (sections == null)
+                return null;
+
+            // Get the .vbn, if it exists
+            var DiscGuardSection = pex.ReadRawSection(".vbn");
+            if (DiscGuardSection != null)
+            {
+                var matchers = new List<ContentMatchSet>
+                {
+                    // Found in "T29.dll" (Redump entry 31914).
+                    // This check should be as long as the following check, as this data is nearly identical (including length) in the original files, but for some reason the section ends early, causing part of the remaining data to not be part of a section.
+                    new ContentMatchSet(new byte?[]
```

Comment:
If this is a subset of the other check, is the other check needed? We have a few places where the longer check is commented out until we can prove it's needed.

Reply:
It is not a subset of the other check, they are subtly different from one another. The fifth byte in the sequence is the first differing one.